app/src/main/java/com/github/houseorganizer/houseorganizer/house/HouseSelectionActivity.java

```diff
@@ -32,21 +41,77 @@
 public class HouseSelectionActivity extends AppCompatActivity {
 
     public static final String HOUSEHOLD_TO_EDIT = "com.github.houseorganizer.houseorganizer.HOUSEHOLD_TO_EDIT";
-    private String emailUser;
-    private RecyclerView housesView;
+    public static final int DEFAULT_UPDATE_INTERVAL = 30;
+    private static final int PERMISSIONS_FINE_LOCATION = 99;
+
+    String emailUser;
+    RecyclerView housesView;
     FirestoreRecyclerAdapter<HouseModel, HouseViewHolder> adapter;
     FirebaseFirestore firestore;
 
+    // Coordinates
+    Double lat;
+    Double lon;
+
+    // Config file for all settings related to FusedLocationProviderClient
+    LocationRequest locationRequest;
+    // Google's API for location services
+    FusedLocationProviderClient fusedLocationProviderClient;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_house_selection);
 
         housesView = findViewById(R.id.housesView);
-
         emailUser = Objects.requireNonNull(FirebaseAuth.getInstance().getCurrentUser()).getEmail();
-        this.firestore = FirebaseFirestore.getInstance();
+        firestore = FirebaseFirestore.getInstance();
+        locationRequest = LocationRequest
+                .create()
+                .setInterval(1000 * DEFAULT_UPDATE_INTERVAL)
+                .setPriority(LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY);
+
+        requestPermission();
+        getCoordinates();
+        setHousesView();
+    }
+
+    private void requestPermission() {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            // Permissions not granted yet
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Cirrus says that this check is not necessary as "SDK_INT is always >= 26" but maybe there is something else ^^

Reply:
True, I don't think it is necesary. I will remove it